cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/SymbolResolver.kt

```diff
@@ -473,6 +474,11 @@ open class SymbolResolver(ctx: TranslationContext) : ComponentPass(ctx) {
             is Reference -> handleReference(currClass, node)
             is ConstructExpression -> handleConstructExpression(node)
             is CallExpression -> handleCallExpression(scopeManager.currentRecord, node)
+
+            // the following function are more for type-resolving, but they are placed inside the
+            // symbol resolver, because this allows us to have the necessary location of nodes where
+            // we infer from
+            is RecordDeclaration -> handleRecordDeclaration(node)
```

Comment:
I need to check, I think this is not necessary anymore, this should already be handled by the type resolver now.